test/e2e/virtual/syncer/virtualworkspace_test.go

```diff
@@ -707,6 +709,245 @@ func TestSyncerVirtualWorkspace(t *testing.T) {
+				if err != nil {
+					t.Logf("error seen waiting for services crd to become active: %v", err)
+					return false
+				}
+				return true
+			}, wait.ForeverTestTimeout, time.Millisecond*100)
+
+			t.Log("Waiting for the persistentvolumes crd to be imported and available in the kubelike source cluster...")
+			require.Eventually(t, func() bool {
+				_, err := kubeClusterClient.CoreV1().PersistentVolumes().Cluster(kubelikeWorkspace).List(ctx, metav1.ListOptions{})
+				if err != nil {
+					t.Logf("error seen waiting for persistentvolumes crd to become active: %v", err)
+					return false
+				}
+				return true
+			}, wait.ForeverTestTimeout, time.Millisecond*100)
+
+			syncTargetKey := workloadv1alpha1.ToSyncTargetKey(kubelikeSyncer.SyncerConfig.SyncTargetWorkspace, kubelikeSyncer.SyncerConfig.SyncTargetName)
+
+			t.Log("Creating a persistentvolume in the kubelike source cluster...")
+			pv := &corev1.PersistentVolume{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-pv",
+					Labels: map[string]string{
+						"state.workload.kcp.dev/" + syncTargetKey: "Upsync",
+					},
+				},
+				Spec: corev1.PersistentVolumeSpec{
+					AccessModes: []corev1.PersistentVolumeAccessMode{
+						corev1.ReadWriteOnce,
+					},
+					Capacity: corev1.ResourceList{
+						corev1.ResourceStorage: resource.MustParse("1Gi"),
+					},
+					PersistentVolumeSource: corev1.PersistentVolumeSource{
+						HostPath: &corev1.HostPathVolumeSource{
+							Path: "/tmp/data",
+						},
+					},
+				},
+			}
+			t.Logf("Creating PV %s through upsyncer virtual workspace...", pv.Name)
+			_, err = kubeClusterClient.CoreV1().PersistentVolumes().Cluster(kubelikeWorkspace).Create(ctx, pv, metav1.CreateOptions{})
+			require.NoError(t, err)
+
+			// create virtual workspace rest configs
+			rawConfig, err := server.RawConfig()
+			require.NoError(t, err)
+			virtualWorkspaceRawConfig := rawConfig.DeepCopy()
+			virtualWorkspaceRawConfig.Clusters["kubelike"] = rawConfig.Clusters["base"].DeepCopy()
+			virtualWorkspaceRawConfig.Clusters["kubelike"].Server = rawConfig.Clusters["base"].Server + "/services/upsyncer/" + kubelikeWorkspace.String() + "/" + kubelikeSyncer.SyncerConfig.SyncTargetName + "/" + kubelikeSyncer.SyncerConfig.SyncTargetUID
+			virtualWorkspaceRawConfig.Contexts["kubelike"] = rawConfig.Contexts["base"].DeepCopy()
+			virtualWorkspaceRawConfig.Contexts["kubelike"].Cluster = "kubelike"
+			kubelikeVWConfig, err := clientcmd.NewNonInteractiveClientConfig(*virtualWorkspaceRawConfig, "kubelike", nil, nil).ClientConfig()
+			kubelikeVWConfig = kcpclienthelper.SetMultiClusterRoundTripper(rest.AddUserAgent(rest.CopyConfig(kubelikeVWConfig), t.Name()))
+			require.NoError(t, err)
+
+			t.Log("Starting test...")
+			testCase.work(t, kubelikeVWConfig, kubelikeWorkspace, syncTargetKey)
+		})
+	}
+}
+
 func toYAML(t *testing.T, obj interface{}) string {
```

Comment:
Please delete this and any uses of it, xref #2273

Reply:
done